File: src/client/flutter/lib/models/entities/town.dart
class Town {
  String codeInsee;
  String name;
  String zipCode;
  int? nbResident;
  String codeDept;

  Town({
    required this.codeInsee,
    required this.name,
    required this.zipCode,
    this.nbResident,
    required this.codeDept,
  });

  Town.fromJson(Map<String, dynamic> json)
      : codeInsee = json["code_insee"],
        name = json["name"],
        zipCode = json["zip_code"],
        codeDept = json["department_code"];

  @override
  String toString(){
    return "$name ($zipCode)";
  }

}

File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/Town.kt
package com.cyberacy.app.models.entities

import com.cyberacy.app.models.services.ApiConnection
import com.google.gson.annotations.SerializedName
import retrofit2.HttpException
import retrofit2.await

class Town(

    @SerializedName("code_insee")
    val codeInsee: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("zip_code")
    val zip_code: String,

    @SerializedName("nb_resident")
    val nbResident: Int,

    @SerializedName("department_code")
    val deptCode: Int,

) {

    override fun toString(): String {
        return "$name ($zip_code)"
    }

    companion object Service {

        suspend fun getTownsFromDept(codeDept: String): List<Town> {
            try {
                val response: ResponseAPI<List<Town>> =
                    ApiConnection.connection().getTownsFromDepartment(codeDept).await()
                return response.data ?: emptyList()
            } catch (e: HttpException) {
                throw e
            }
        }

    }

}